node/worker/worker.go

```diff
@@ -92,107 +96,97 @@ func (w *Worker) throttleTxs(selected types.Transactions, recentTxsStats types.R
+			utils.Logger().Info().Str("sender", sender).Msg("Gas limit exceeded for current block")
+			txs.Pop()
+
+		case core.ErrNonceTooLow:
+			// New head notification data race between the transaction pool and miner, shift
+			utils.Logger().Info().Str("sender", sender).Uint64("nonce", tx.Nonce()).Msg("Skipping transaction with low nonce")
+			txs.Shift()
+
+		case core.ErrNonceTooHigh:
+			// Reorg notification data race between the transaction pool and miner, skip account =
+			utils.Logger().Info().Str("sender", sender).Uint64("nonce", tx.Nonce()).Msg("Skipping account with high nonce")
+			txs.Pop()
+
+		case nil:
+			// Everything ok, collect the logs and shift in the next transaction from the same account
+			coalescedLogs = append(coalescedLogs, logs...)
+			txs.Shift()
+
+		default:
+			// Strange error, discard the transaction and get the next in line (note, the
+			// nonce-too-high clause will prevent us from executing in vain).
+			utils.Logger().Info().Str("hash", tx.Hash().Hex()).AnErr("err", err).Msg("Transaction failed, account skipped")
+			txs.Shift()
 		}
-
-		utils.Logger().Info().Str("txId", tx.Hash().Hex()).Uint64("txGasLimit", tx.Gas()).Msg("Transaction gas limit info")
-	}
-
-	utils.Logger().Info().Uint64("newBlockNum", newBlockNum).Int("newTxns", len(selected)).Uint64("blockGasLimit", w.current.header.GasLimit()).Uint64("blockGasUsed", w.current.header.GasUsed()).Msg("Block gas limit and usage info")
-
-	return selected, unselected, invalid
-}
-
-// SelectStakingTransactionsForNewBlock selects staking transactions for new block.
-func (w *Worker) SelectStakingTransactionsForNewBlock(
-	newBlockNum uint64, txs staking.StakingTransactions,
-	coinbase common.Address) (staking.StakingTransactions, staking.StakingTransactions, staking.StakingTransactions) {
-
-	// only beaconchain process staking transaction
-	if w.chain.ShardID() != values.BeaconChainShardID {
-		utils.Logger().Warn().Msgf("Invalid shardID: %v", w.chain.ShardID())
-		return nil, nil, nil
 	}
 
-	if w.current.gasPool == nil {
-		w.current.gasPool = new(core.GasPool).AddGas(w.current.header.GasLimit())
-	}
-
-	selected := staking.StakingTransactions{}
-	// TODO: chao add total gas fee checking when needed
-	unselected := staking.StakingTransactions{}
-	invalid := staking.StakingTransactions{}
-	for _, tx := range txs {
-		snap := w.current.state.Snapshot()
-		_, err := w.commitStakingTransaction(tx, coinbase)
-		if err != nil {
-			w.current.state.RevertToSnapshot(snap)
-			invalid = append(invalid, tx)
-			utils.Logger().Error().Err(err).Str("stakingTxId", tx.Hash().Hex()).Msg("Commit staking transaction error")
-		} else {
-			selected = append(selected, tx)
-			utils.Logger().Info().Str("stakingTxId", tx.Hash().Hex()).Uint64("txGasLimit", tx.Gas()).Msg("StakingTransaction gas limit info")
+	// STAKING - only beaconchain process staking transaction
+	if w.chain.ShardID() == values.BeaconChainShardID {
```

Comment:
right now, this part is no-op right? we just log the coalescedLogs here?

Reply:
yes.